contracts/src/protocol/UBIPool.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.23;
+
+import { Ownable2StepUpgradeable } from "@openzeppelin/contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
+import { ReentrancyGuardUpgradeable } from "@openzeppelin/contracts-upgradeable/utils/ReentrancyGuardUpgradeable.sol";
+import { PubKeyVerification } from "./PubKeyVerification.sol";
+import { IUBIPool } from "../interfaces/IUBIPool.sol";
+
+/// @title UBIPool
+/// @notice Contract for distributing UBI to validators
+/// UBI comes from a percentage of the protocol's emmission during a defined period
+/// This % is minted to the UBIPool contract and can be claimed by validators
+/// Each validator can claim their UBI for a given month
+/// Distributions will be made public monthly by Story for community scrutiny,
+/// and if correct, will be distributed to validators via this contract
+contract UBIPool is IUBIPool, Ownable2StepUpgradeable, ReentrancyGuardUpgradeable, PubKeyVerification {
+    /// @notice The maximum UBI percentage
+    uint32 public immutable MAX_UBI_PERCENTAGE;
```

Comment:
What's the decimal exponent for this (since Solidity doesn't support decimals)? 100% => 100_0000? Need to ensure we have enough accuracy in UBI %

Reply:
I was using 2 decimals. What's the preferred amount by CL @ezreal1997 ?

Reply:
2 decimals is fine for this